src/backend/distributed/commands/function.c

```diff
@@ -1303,18 +1287,92 @@ PostprocessCreateFunctionStmt(Node *node, const char *queryString)
 		return NIL;
 	}
 
-	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
-	EnsureDependenciesExistOnAllNodes(&address);
+	ObjectAddress functionAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
 
-	List *commands = list_make4(DISABLE_DDL_PROPAGATION,
-								GetFunctionDDLCommand(address.objectId, true),
-								GetFunctionAlterOwnerCommand(address.objectId),
-								ENABLE_DDL_PROPAGATION);
+	if (IsObjectAddressOwnedByExtension(&functionAddress, NULL))
+	{
+		return NIL;
+	}
+
+	ObjectAddress *undistributableDependency = GetUndistributableRelationDependency(
+		&functionAddress);
+	if (undistributableDependency != NULL)
+	{
+		if (SupportedDependencyByCitus(undistributableDependency))
+		{
+			RangeVar *functionRangeVar = makeRangeVarFromNameList(stmt->funcname);
+			char *functionName = functionRangeVar->relname;
+			char *dependentRelationName =
+				get_rel_name(undistributableDependency->objectId);
+
+			ereport(WARNING, (errmsg("Citus can't distribute function \"%s\" having "
```

Comment:
can't we only have errhint differ:
```
if (...)
{
  char errorHint = SupportedDependencyByCitus(undistributableDependency) ? "...." : "..."

.}
```

Reply:
Since we can't distribute views as of now, providing that info on the errmsg makes more sense to me.